internal/server/v1/firehose/envvars.go

```diff
@@ -21,7 +21,7 @@ func (api *firehoseAPI) buildEnvVars(ctx context.Context, firehose *models.Fireh
 	}
 
 	firehose.Configs.EnvVars["SCHEMA_REGISTRY_STENCIL_ENABLE"] = trueString
-	if fetchStencilURL || firehose.Configs.EnvVars[configStencilURL] == "" {
+	if firehose.Configs.EnvVars[configStencilURL] == "" && !skipStencil {
```

Comment:
are we going to skip stencilURL creation even if `firehose.Configs.EnvVars[configStencilURL]` value is empty?

Reply:
yes this is to keep current behaviour on PATCH api where we only want to update certain fields instead of PUT calls where we update all fields.

although it is open for discussion what's the best way, but I was thinking to keep the current behaviour first.

Reply:
it makes sense to maintain the existing flow of PATCH and PUT we need this, LGTM.